include/highfive/H5DataType.hpp

```diff
@@ -217,11 +217,11 @@ class CompoundType: public DataType {
     /// \brief Initializes a compound type from a vector of member definitions
     /// \param t_members
     /// \param size
-    inline CompoundType(const std::vector<member_def>& t_members, size_t size = 0)
+    inline explicit CompoundType(const std::vector<member_def>& t_members, size_t size = 0)
```

Comment:
Undecided: member_def is tightly linked to CompoundType. Hence, maybe not worth breaking user code over something that's unlikely to be very surprising.

Reply:
I'll revert, not worth breaking anyone's code over this.